src/springfield/fields.py

```diff
@@ -287,12 +287,67 @@ def adapt(self, value):
                 return unicode(value)
             raise
 
+
 class BytesField(AdaptableTypeField):
     """
     A :class:`Field` that contains binary bytes (or a legacy Python 2 string).
+
+    Can optionally specify an encoding to use for json/unicode conversion, such
+    as 'hex' or 'base64' if the values are non-ASCII this might be more readable
+    than a bunch of unicode escapes.
     """
 
     type = bytes
+    encoding = None
+
+    def __init__(self, encoding=None, *args, **kwargs):
+        """
+        Construct a Bytes field
+
+        :param encoding: Optional encoding to use for jsonify(), such as
+           'hex' or 'base64'
+        """
+        super(BytesField, self).__init__(*args, **kwargs)
+        self.encoding = encoding
+
+    def jsonify(self, value):
+        """
+        Encode the bytes into a unicode string suitable for json encoding.
+
+        If an encoding was specified for the field, it is applied here.
+        """
+        if not isinstance(value, bytes):
+            raise ValueError('BytesField must contain bytes')
+
+        # Apply hex/base64 encoding if desired
+        if self.encoding:
+            value = value.encode(self.encoding)
+
+        # Convert to unicode using an 8-bit encoding to retain binary data
+        return value.decode('latin1')
```

Comment:
With `latin1`, people can just pass-through their python strings without any encoding/decoding at all, and let JSON itself do the escaping, which I think may have a use case occasionally, like if the bytes are actually some sort of ASCII blob (like PEM formatted certificates, stuff like that).